digiroad2-api-oth/src/test/scala/fi/liikennevirasto/digiroad2/dataimport/CsvDataImporterSpec.scala

```diff
@@ -388,6 +386,166 @@ class CsvDataImporterSpec extends AuthenticatedApiSpec with BeforeAndAfter {
         csvRow = trafficSignCsvImporter.rowToString(defaultValues ++ assetFields)))))
   }
 
+  test("validation for traffic sign import with sign roadwork and dates", Tag("db")) {
+    val assetFields = Map("koordinaatti x" -> 1, "koordinaatti y" -> 1, "liikennemerkin tyyppi" -> "A11", "suuntima" -> "40",
+      "alkupaivamaara" -> "03.08.2020", "loppupaivamaara" -> "07.08.2020", "kunnan id" -> "408")
+
+    val validCsv = csvToInputStream(createCsvForTrafficSigns(assetFields))
+
+    runWithRollback {
+      when(mockRoadLinkService.getClosestRoadlinkForCarTrafficFromVVH(any[User], any[Point], any[Boolean])).thenReturn(roadLink)
+      when(mockRoadLinkService.filterRoadLinkByBearing(any[Option[Int]], any[Option[Int]], any[Point], any[Seq[RoadLink]])).thenReturn(roadLink)
+
+      val result = trafficSignCsvImporter.processing(validCsv, Set(), testUser)
+
+      result.incompleteRows.size should be(0)
+      result.malformedRows.size should be(0)
+      result.excludedRows.size should be(0)
+      result.notImportedData.size should be(0)
+      result.createdData.size should be(1)
+
+      val startDate = result.createdData.head.properties.find(_.columnName == "startDate").get match {
+        case v => v.value
+        case _ => ""
+      }
+
+      val endDate = result.createdData.head.properties.find(_.columnName == "endDate").get match {
+        case v => v.value
+        case _ => ""
+      }
+
+      startDate should be("03.08.2020")
+      endDate should be("07.08.2020")
```

Comment:
We can avoid all of this effort by doing the comparations directly:
```Scala
result.createdData.head.properties.find(_.columnName == "startDate").get should equal(Some("03.08.2020"))
result.createdData.head.properties.find(_.columnName == "endDate").get should equal(Some("07.08.2020"))

Reply:
Done